rename.cc

```diff
@@ -38,10 +43,18 @@ TEST(RenameAt, AbsDesignationSame) {
 
     // Various rename from/to the same absolute path; in each case the source
     // and dest directory FDs should be irrelevant.
-    EXPECT_OK(renameat(AT_FDCWD, src_path, AT_FDCWD, src_path));
-    EXPECT_OK(renameat(AT_FDCWD, src_path, dfd, src_path));
-    EXPECT_OK(renameat(dfd, src_path, AT_FDCWD, src_path));
-    EXPECT_OK(renameat(dfd, src_path, dfd, src_path));
+    CheckRenameat(AT_FDCWD, src_path, AT_FDCWD, src_path);
+    CheckRenameat(dfd, src_path, AT_FDCWD, src_path);
+
+    // Bug 222258 hasn't been fixed on FreeBSD, yet. Once it has a
+    // __FreeBSD_version__ will be filled in here as a guard.
+#if defined(__FreeBSD__)
+    EXPECT_NONFATAL_FAILURE(CheckRenameat(AT_FDCWD, src_path, dfd, src_path), "");
```

Comment:
`EXPECT_NONFATAL_FAILURE` seems a bit gtest-internal, particularly without any match string.  Would it be easier to just have the two tests only run `if !defined(__FreeBSD__)` ?  Or does the non-fatal check add something?

Reply:
Note that the kernel issue has just been fixed, so we will want to run the test on FreeBSD (and XFAIL will be version-specific).

https://reviews.freebsd.org/rS343891

Reply:
> Note that the kernel issue has just been fixed, so we will want to run the test on FreeBSD (and XFAIL will be version-specific).
> 
> https://reviews.freebsd.org/rS343891

I frankly didn't expect the underlying issue to be fixed so quickly :).

I'd prefer to not overcomplicate the test suite with `__FreeBSD_version` checks. If we import the test suite (we can now), we can be more clever about the `__FreeBSD_version` checks there.